arrow-flight/tests/client.rs

```diff
@@ -259,6 +232,248 @@ async fn test_do_get_error_in_record_batch_stream() {
     .await;
 }
 
+#[tokio::test]
+async fn test_do_put() {
+    do_test(|test_server, mut client| async move {
+        client.add_header("foo", "bar").unwrap();
+
+        // encode the batch as a stream of FlightData
+        let input_flight_data = test_flight_data().await;
+
+        let expected_response = vec![
+            PutResult {
+                app_metadata: Bytes::from("foo"),
```

Comment:
```suggestion
                app_metadata: Bytes::from("hello"),
```

Initially I got confused thinking this was matching the header

Reply:
will make the header names more distinct

Reply:
in 01bdadb79